src/postprocess/band_structure.jl

```diff
@@ -19,18 +19,30 @@ function high_symmetry_kpath(model; kline_density=20)
     # https://github.com/louisponet/DFControl.jl/blob/master/src/structure.jl
     # but for this the best way to go would be to refactor into a small "CrystalStructure"
     # julia module which deals with these sort of low-level details everyone can agree on.
-    pystructure = pymatgen_structure(model.lattice, model.atoms)
-    symm_kpath = pyimport("pymatgen.symmetry.bandstructure").HighSymmKpath(pystructure)
-    kcoords, labels = symm_kpath.get_kpoints(kline_density, coords_are_cartesian=false)
-
+    # pystructure = pymatgen_structure(model.lattice, model.atoms)
+    # symm_kpath = pyimport("pymatgen.symmetry.bandstructure").HighSymmKpath(pystructure)
+    # kcoords, labels = symm_kpath.get_kpoints(kline_density, coords_are_cartesian=false)
+    
+    sgnum = spglib_get_spacegroup(model.lattice, model.atoms)
+    Rs = [model.lattice[i, :] for i in 1:size(model.lattice,1)]
+    kp        = irrfbz_path(sgnum, Rs)
+    N         = kline_density*length(vcat(kp.paths...))
+    kcoords   = collect(Brillouin.interpolate(kp, N))
+    
     labels_dict = Dict{String, Vector{eltype(kcoords[1])}}()
-    for (ik, k) in enumerate(kcoords)
-        if length(labels[ik]) > 0
-            labels_dict[detexify_kpoint(labels[ik])] = k
-        end
+    for (key, val) in kp.points
+        labels_dict[string(key)] = val
+    end
+    kpath = []
+    for path in kp.paths
+        new_path = []
+        for el in path
+            push!(new_path, retexify_kpoint(string(el)))
+	end
+        push!(kpath, new_path)
```

Comment:
indentation is a bit off. Also can't you just do `new_path = retexify_kpoint.(string.(path))`?

Reply:
similarly for the kpath above, looks like that logic can be simplified. Generally we're trying to move away from the "push!" idiom